esi/src/expression.rs

```diff
@@ -128,52 +149,74 @@ impl EvalContext {
         self.request = request;
     }
 }
+
 impl<const N: usize> From<[(String, Value); N]> for EvalContext {
     fn from(data: [(String, Value); N]) -> Self {
         Self::new_with_vars(HashMap::from(data))
     }
 }
 
-fn format_key(key: &str, subkey: Option<String>) -> String {
+fn format_key(key: &str, subkey: Option<&str>) -> String {
     subkey.map_or_else(|| key.to_string(), |subkey| format!("{key}[{subkey}]"))
 }
-
-#[derive(Debug, Clone, PartialEq)]
+/// Represents a value in an ESI expression.
+///
+/// Values can be of different types:
+/// - `Integer`: A 32-bit signed integer
+/// - `String`: A UTF-8 string
+/// - `Boolean`: A boolean value (true/false)
+/// - `Null`: Represents an absence of value
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Value {
     Integer(i32),
-    String(String),
-    Boolean(BoolValue),
+    Text(Cow<'static, str>),
```

Comment:
Oh, also why not convert everything to `&str`? Just trying to avoid all the lifetimes?

Reply:
@tyler I'm going to resolve this to get it merged. Please open another PR if you want to improve it.